From 53f6eca921a44a1f5e567b92d45ea84599afea74 Mon Sep 17 00:00:00 2001
From: Rohan Garg <rohangarg@kubuntu.org>
Date: Wed, 19 Sep 2012 00:03:59 +0530
Subject: [PATCH] Use cascaded configs for other modules as well

In addition to commit d0343319fcfc249a38c79171be727d7133984eeb, one also
needs this commit to get cascaded configs to work
---
 powerdevil/kcmodule/activities/activitywidget.cpp |    2 +-
 powerdevil/kcmodule/profiles/EditPage.cpp         |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/powerdevil/kcmodule/activities/activitywidget.cpp b/powerdevil/kcmodule/activities/activitywidget.cpp
index b1b3d7e..8d9517a 100644
--- a/powerdevil/kcmodule/activities/activitywidget.cpp
+++ b/powerdevil/kcmodule/activities/activitywidget.cpp
@@ -32,7 +32,7 @@
 ActivityWidget::ActivityWidget(const QString& activity, QWidget* parent)
     : QWidget(parent)
     , m_ui(new Ui::ActivityWidget)
-    , m_profilesConfig(KSharedConfig::openConfig("powermanagementprofilesrc", KConfig::SimpleConfig))
+    , m_profilesConfig(KSharedConfig::openConfig("powermanagementprofilesrc", KConfig::SimpleConfig | KConfig::CascadeConfig))
     , m_activity(activity)
     , m_activityConsumer(new KActivities::Consumer(this))
     , m_actionEditWidget(new ActionEditWidget(QString("Activities/%1/SeparateSettings").arg(activity)))
diff --git a/powerdevil/kcmodule/profiles/EditPage.cpp b/powerdevil/kcmodule/profiles/EditPage.cpp
index 69a1a33..95a98d9 100644
--- a/powerdevil/kcmodule/profiles/EditPage.cpp
+++ b/powerdevil/kcmodule/profiles/EditPage.cpp
@@ -84,7 +84,7 @@ EditPage::EditPage(QWidget *parent, const QVariantList &args)
 
     setupUi(this);
 
-    m_profilesConfig = KSharedConfig::openConfig("powermanagementprofilesrc", KConfig::SimpleConfig);
+    m_profilesConfig = KSharedConfig::openConfig("powermanagementprofilesrc", KConfig::SimpleConfig | KConfig::CascadeConfig);
 
     if (m_profilesConfig->groupList().isEmpty()) {
         // Use the generator
-- 
1.7.10.4

